cpp/src/arrow/array/array_union_test.cc

```diff
@@ -70,6 +70,22 @@ TEST(TestUnionArray, TestSliceEquals) {
   CheckUnion(batch->column(1));
 }
 
+TEST(TestSparseUnionArray, ValidateFullNullable) {
+  auto ty = sparse_union(
+      {field("ints", int64()), field("strs", utf8(), /*nullable*/ false)}, {2, 7});
+  auto ints = ArrayFromJSON(int64(), "[0, 1, 2, 3]");
+  auto strs = ArrayFromJSON(utf8(), R"(["a", null, "c", "d"])");
```

Comment:
What if instead I have `[null, "a", "c", "d"]`? Does it validate? In this case the null wouldn't be part of the parent (logical) array....

Reply:
Yes, it did validate

Reply:
I think this makes sense? A union is always non-nullable